### PR TITLE
fix issue 1116

### PR DESCRIPTION
**Pull request by Yang2096**
I checked every `(','? ......)+` and `(','? ......)*` in `PlSqlParser.g4`, hoping this would fix issue #1116 .

### CODE REVIEW

**Comment by KvanTTT** [comment]
Could you please add a code sample that covers this fix?

**Comment by Yang2096** [comment]
> 
> 
> Could you please add a code sample that covers this fix?

under the `examples` directory? Sorry that i don't konw how to prove my PR works.
I think it is impossible to write a code sample that is valid in the grammar i just wrote and invalid in the former grammar, due to every sql works with
```
comma_separated_list
    : list_item ( ',' list_item)*
    ;
```
, works with
```
comma_separated_list
    : (','? list_item)+
    ;
```
too.

File: plsql/PlSqlParser.g4 [hunk 1]
```diff
@@ -648,41 +648,48 @@ local_partitioned_index
...
-    : '(' (','? PARTITION partition_name?
-              ((segment_attributes_clause | key_compression)+ )?
-              UNUSABLE? )+
-      ')'
+    : '(' partitioned_table (',' partitioned_table)* ')'
+    ;
+
+partitioned_table
+    :  PARTITION partition_name?
+        ((segment_attributes_clause | key_compression)+ )?
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Simplify?

```g4
(segment_attributes_clause | key_compression)*
```

File: plsql/PlSqlParser.g4 [hunk 2]
```diff
@@ -648,41 +648,48 @@ local_partitioned_index
...
-            ((segment_attributes_clause | key_compression)+)?
-            UNUSABLE index_subpartition_clause? )+
-        ')'
+    : (STORE IN '(' tablespace (',' tablespace)* ')' )?
+        '(' on_comp_partitioned_clause (',' on_comp_partitioned_clause)* ')'
+    ;
+
+on_comp_partitioned_clause
+    : PARTITION partition_name?
+        ((segment_attributes_clause | key_compression)+)?
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]:
```g4
(segment_attributes_clause | key_compression)*
```

File: plsql/PlSqlParser.g4 [hunk 3]
```diff
@@ -1802,13 +1809,12 @@ relational_table
     ;
 
 relational_properties
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Replace with `relational_property` (singular form)

File: plsql/PlSqlParser.g4 [hunk 4]
```diff
@@ -2071,13 +2077,14 @@ log_grp
     ;
 
 supplemental_table_logging
-    : ADD
-       (','? SUPPLEMENTAL LOG  (supplemental_log_grp_clause | supplemental_id_key_clause) )*
-    | DROP (','? SUPPLEMENTAL LOG (supplemental_id_key_clause | GROUP log_grp) )*
+    : ADD (SUPPLEMENTAL LOG  (supplemental_log_grp_clause | supplemental_id_key_clause) )?
```
Comment (KvanTTT) [inline plsql/PlSqlParser.g4]: Are you sure that `?` is required here?

Comment (Yang2096) [reply plsql/PlSqlParser.g4]: Yes, `(','? xxx)*` should be changed to `xxx? (',' xxx)*`

File: plsql/PlSqlParser.g4 (hunk 3 above)
Comment (Yang2096) [reply plsql/PlSqlParser.g4]: ok

File: plsql/PlSqlParser.g4 (hunk 4 above)
Comment (Yang2096) [reply plsql/PlSqlParser.g4]: But now i see there is a mistake with the former grammar, it should be `(','? xxx)+`. That `?` will be removed.

Comment (KvanTTT) [reply plsql/PlSqlParser.g4]: I think it's better to replace `(','? xxx)*` with `(xxx (',' xxx)*)?`

**Comment by KvanTTT** [comment]
Thanks. Now it seems that everything is fine.

File: plsql/PlSqlParser.g4 [hunk 5]
```diff
@@ -4294,16 +4303,16 @@ keep_clause
     ;
 
 function_argument
-    : '(' (','? argument)* ')' keep_clause?
+    : '(' argument? (',' argument)* ')' keep_clause?
```
Comment (Yang2096) [inline plsql/PlSqlParser.g4]: @KvanTTT   I fixed this in a wrong way, `(','? argument)*` must be replaced with `(xxx (',' xxx)*)?`

**Pull request merged**